File: VcheckSDKDemoIOS/domain/document_verification_codes.kt
package com.vcheck.demo.dev.domain

enum class DocumentVerificationCode {
    OK, //0
    InvalidPagesCount,  //1 - Неверное кол-во загруженных файлов
    InvalidVerificationStage,  //2 - Этап заявки не соответствует разрешенному для данного запроса
    UploadAttemptsExceeded,  //3 - Кол-во попыток на загрузку истекло
    InvalidFiles, //4 - Неверное расширение документа
    InvalidDocumentType, //5 - Неверный тип документа
    PrimaryAlreadyExists,  //6 - Основой документ уже существует
    PhotoTooLarge, //7 - Вес файлов превышает допустимый
}

fun DocumentVerificationCode.toCodeIdx(): Int {
    return when(this) {
        DocumentVerificationCode.OK -> 0
        DocumentVerificationCode.InvalidPagesCount -> 1
        DocumentVerificationCode.InvalidVerificationStage -> 2
        DocumentVerificationCode.UploadAttemptsExceeded -> 3
        DocumentVerificationCode.InvalidFiles -> 4
        DocumentVerificationCode.InvalidDocumentType -> 5
        DocumentVerificationCode.PrimaryAlreadyExists -> 6
        DocumentVerificationCode.PhotoTooLarge -> 7
    }
}

fun codeIdxToVerificationCode(codeIdx: Int)
    : DocumentVerificationCode {
    return when(codeIdx) {
        0 -> DocumentVerificationCode.OK
        1 -> DocumentVerificationCode.InvalidPagesCount
        2 -> DocumentVerificationCode.InvalidVerificationStage
        3 -> DocumentVerificationCode.UploadAttemptsExceeded
        4 -> DocumentVerificationCode.InvalidFiles
        5 -> DocumentVerificationCode.InvalidDocumentType
        6 -> DocumentVerificationCode.PrimaryAlreadyExists
        else -> DocumentVerificationCode.PhotoTooLarge
    }
}

File: VcheckSDKDemoIOS/domain/DocumentVerificationCodes.swift
//
//  DocumentVerificationCodes.swift
//  VcheckSDKDemoIOS
//
//  Created by Kirill Kaun on 06.05.2022.
//

import Foundation

enum DocumentVerificationCode {
    case OK //0
    case InvalidPagesCount  //1 - Неверное кол-во загруженных файлов
    case InvalidVerificationStage  //2 - Этап заявки не соответствует разрешенному для данного запроса
    case UploadAttemptsExceeded  //3 - Кол-во попыток на загрузку истекло
    case InvalidFiles //4 - Неверное расширение документа
    case InvalidDocumentType //5 - Неверный тип документа
    case PrimaryAlreadyExists  //6 - Основой документ уже существует
    case PhotoTooLarge //7 - Вес файлов превышает допустимый
}

extension DocumentVerificationCode {
    func toCodeIdx() -> Int {
        switch(self) {
            case DocumentVerificationCode.OK: return 0
            case DocumentVerificationCode.InvalidPagesCount: return 1
            case DocumentVerificationCode.InvalidVerificationStage: return 2
            case DocumentVerificationCode.UploadAttemptsExceeded: return 3
            case DocumentVerificationCode.InvalidFiles: return 4
            case DocumentVerificationCode.InvalidDocumentType: return 5
            case DocumentVerificationCode.PrimaryAlreadyExists: return 6
            case DocumentVerificationCode.PhotoTooLarge: return 7
        }
    }
}

func codeIdxToVerificationCode(codeIdx: Int) -> DocumentVerificationCode {
    switch(codeIdx) {
        case 0: return DocumentVerificationCode.OK
        case 1: return DocumentVerificationCode.InvalidPagesCount
        case 2: return DocumentVerificationCode.InvalidVerificationStage
        case 3: return DocumentVerificationCode.UploadAttemptsExceeded
        case 4: return DocumentVerificationCode.InvalidFiles
        case 5: return DocumentVerificationCode.InvalidDocumentType
        case 6: return DocumentVerificationCode.PrimaryAlreadyExists
        default: return DocumentVerificationCode.PhotoTooLarge
    }
}


